src/lib.rs

```diff
@@ -913,11 +913,15 @@ impl Connection {
 
                     // Feed crypto data to the TLS state, if there's data
                     // available at the expected offset.
-                    if space.crypto_stream.readable() {
-                        let buf = space.crypto_stream.recv_pop(std::usize::MAX)?;
-                        let level = space.crypto_level;
+                    let mut crypto_buf: [u8; 512] = [0; 512];
 
-                        self.tls_state.provide_data(level, &buf)
+                    let level = space.crypto_level;
```

Comment:
Just a note on this change: I had trouble using the `while let` semantics and checking the `readable` state of the stream.  Basically, this doesn't work:
```
while space.crypto_stream.readable() &&
         let Ok((read, _)) = space.crypto_stream.recv_pop(&mut crypto_buf) {
...
```
But the compiler won't let me as the `let` stops me using other boolean operations in the `while`, very rude!

So what I did was add a `return Err(Error::Done)` in `RecvBuf::pop`, and this is something that wasn't done before.

Given that the whole `recv_pop` -> `RecvBuf::pop` stack was already equipped to return `Result` types, I felt this was a good way to deal with this `while let` problem.

Reply:
Yeah, that seems like a good change. The documentation said that we would return `Done` if there was nothing to read, but that was kind of a lie, so this is better now.

Reply:
Actually nvm, the public `stream_recv()` checks that `readable()` is true, so the documentation was accurate. Still, this is a good change for internal use.